### ISSUE TITLE
JobValid functions are never called?

### ISSUE THREAD

**Issue by zxcware** [volcano-sh/volcano#2946]
**What happened**:

Hi team, I was digging into the latest scheduler codes. When I walked through the `OpenSession` function of the framework, I found a seeming bug. 

Look at the following snippet. At line 31, `openSession`, in which JobValid functions will be called, is called to create a session object. However, only after line 45 does a plugin get created and all its JobValid functions, if any, get registered into the session. Does this mean JobValid functions are never called?
<img width="637" alt="image" src="https://github.com/volcano-sh/volcano/assets/2355051/0910e7b7-b417-4538-820b-b36b66b540d3">

**Comment by lowang-bh** [comment]
Yes, you are right. Plugins func only takes effect after it is registry

**Comment by renwenlong-github** [comment]
![image](https://github.com/volcano-sh/volcano/assets/33648707/c27a3c85-c15a-4efe-b45b-d5c89c082be3)
 JobValid functions are called in [ allocate.go backfill.go preempt.go reclaim.go]actions and session.go, Even if it is not called in the open session, it does not affect the judgment of other actions, but the overall scheduling is not so accurate

**Comment by zxcware** [comment]
> 

Nice find! And plus 1 on the point that "scheduling is not so accurate". It doesn't seem to be necessary to call `JobValid` that many times..
